### PR TITLE
react-native: add some missing props for TextInput

### PR DESCRIPTION
**Pull request by Naturalclar**
Added `passwordRules` and `rejectResponderTermination` props for `TextInput` both props are for iOS.

Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/facebook/react-native/blob/master/Libraries/Components/TextInput/TextInput.js#L270-L285

### CODE REVIEW

File: types/react-native/index.d.ts [hunk 1]
```diff
@@ -1075,10 +1090,13 @@ export interface TextInputIOSProps {
     | 'newPassword'
     | 'oneTimeCode';
 
+
```
Comment (TheSavior) [inline types/react-native/index.d.ts]: Unnecessary blank lines?

Comment (alloy) [reply types/react-native/index.d.ts]: Yup, this is causing CI to fail.

File: types/react-native/index.d.ts [hunk 2]
```diff
@@ -990,6 +990,21 @@ export interface TextInputIOSProps {
      */
     keyboardAppearance?: 'default' | 'light' | 'dark';
 
+    /**
+     * Provide rules for your password.
+     * For example, say you want to require a password with at least eight characters consisting of a mix of uppercase and lowercase letters, at least one number, and at most two consecutive characters.
+     * "required: upper; required: lower; required: digit; max-consecutive: 2; minlength: 8;"
+     */
+    passwordRules?: string;
```
Comment (alloy) [inline types/react-native/index.d.ts]:
```suggestion
    passwordRules?: string | null;
```

File: types/react-native/index.d.ts [hunk 3]
```diff
@@ -990,6 +990,21 @@ export interface TextInputIOSProps {
...
+    
+    /**
+     * If `true`, allows TextInput to pass touch events to the parent component.
+     * This allows components to be swipeable from the TextInput on iOS,
+     * as is the case on Android by default.
+     * If `false`, TextInput always asks to handle the input (except when disabled).
+     */
+    rejectResponderTermination?: boolean;
```
Comment (alloy) [inline types/react-native/index.d.ts]:
```suggestion
    rejectResponderTermination?: boolean | null;
```

**Review by alloy — CHANGES_REQUESTED** [review]
Some small `Maybe` changes and CI failure, but otherwise 👍

File: types/react-native/index.d.ts (hunk 1 above)
Comment (Naturalclar) [reply types/react-native/index.d.ts]: oops, I don't know how that snuck in there 😅 
I've removed the blank lines, thanks!

**Comment by alloy** [comment]
Unsure how this didn’t surface earlier, but indeed this test file of another RN package has a syntax error: https://github.com/DefinitelyTyped/DefinitelyTyped/blob/28ff2e3a70ee7172902b2efb0f0c527e8ada0430/types/react-native-popup-dialog/react-native-popup-dialog-tests.tsx#L92

Can you fix that in your PR?

**Comment by Naturalclar** [comment]
@alloy sure! I've fixed the syntax error in react-native-popup-dialog

**Comment by alloy** [comment]
Great, thanks! ✅

**Comment by sandersn** [comment]
@alloy Turns out [we just merged a new syntax error for JSX](https://github.com/microsoft/TypeScript/pull/36636) yesterday. It caught me by surprise too.

**Comment by alloy** [comment]
@sandersn Well, seems to work great! ✨

**Pull request merged**